File: IOSwiftUI/Common/Messages/Images/BOImagesUpdateRequestModel.cs
using System;
using System.ComponentModel.DataAnnotations;
using IOSwiftUI.Common.Enumerations;
using IOSwiftUI.Common.Messages.Base;

namespace IOSwiftUI.Common;

public class BOImagesUpdateRequestModel : RequestModel
{

    [Required]
    public int ID { get; set; }

    public string FileName { get; set; }

    public DateTimeOffset CreateDate { get; set; }

    public int Width { get; set; }

    public int Height { get; set; }

    public bool IsDraft { get; set; }

    public double Price { get; set; }

    public Currencies PriceCurrency { get; set; }

    public int SaleAmount { get; set; }


}

File: IOSwiftUI/Backoffice.UI/src/screens/memberimages/models/BOImagesUpdateRequestModel.ts
import BaseRequestModel from "../../../common/models/BaseRequestModel";
import Currencies from "../enumerations/Currencies";


class BOImagesUpdateRequestModel extends BaseRequestModel {

    id: number | null;
    fileName: string | null;
    createDate: string | null;
    width: number | null;
    height: number | null;
    isDraft: boolean;
    price: number | null;
    priceCurrency: Currencies;
    saleAmount: number | null;

    constructor() {
        super();

        this.id = null;
        this.fileName = null;
        this.createDate = null;
        this.width = null;
        this.height = null;
        this.isDraft = false;
        this.price = null;
        this.priceCurrency = Currencies.USDollar;
        this.saleAmount = null;

    }
}

export default BOImagesUpdateRequestModel;
